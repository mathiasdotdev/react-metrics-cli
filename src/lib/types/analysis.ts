/**
 * Type de déclaration détectée dans le code
 */
export enum DeclarationType {
  CONSTANT = 'CONSTANT',
  FUNCTION = 'FUNCTION',
  CLASS = 'CLASS',
  CONSOLE = 'CONSOLE',
  PROP = 'PROP',
  EXPORT = 'EXPORT',
  DEPENDENCY = 'DEPENDENCY',
  DEFINITION = 'DEFINITION',
}

/**
 * Localisation d'une déclaration dans un fichier
 */
export interface Location {
  filePath: string;
  line: number;
  column: number;
}

/**
 * Représente une déclaration trouvée dans le code
 */
export interface Declaration {
  name: string;
  type: DeclarationType;
  location: Location;
  isUsedLocally: boolean;
  isImportedExternally: boolean;
  isDeprecated?: boolean;
  context?: string;
  metadata?: Record<string, unknown>;
}

/**
 * Résultat complet de l'analyse
 */
export interface AnalysisResult {
  declarations: Map<string, Declaration>;
  deadCode: Declaration[];
  deprecated: Declaration[];
  duration: number;
}

/**
 * Génère une clé unique pour une déclaration
 */
export const getDeclarationKey = (declaration: Declaration): string => {
  return `${declaration.location.filePath}:${declaration.location.line}:${declaration.location.column}:${declaration.name}`;
};

/**
 * Vérifie si une déclaration est utilisée (localement ou importée ailleurs)
 */
export const isUsed = (declaration: Declaration): boolean => {
  return declaration.isUsedLocally || declaration.isImportedExternally;
};

/**
 * Récupère le code mort (déclarations non utilisées)
 */
export const getDeadCode = (
  declarations: Map<string, Declaration>,
): Declaration[] => {
  return Array.from(declarations.values()).filter((d) => !isUsed(d));
};

/**
 * Récupère toutes les déclarations marquées @deprecated
 */
export const getDeprecated = (
  declarations: Map<string, Declaration>,
): Declaration[] => {
  return Array.from(declarations.values())
    .filter((d) => d.isDeprecated === true)
    .sort((a, b) => a.location.filePath.localeCompare(b.location.filePath));
};
