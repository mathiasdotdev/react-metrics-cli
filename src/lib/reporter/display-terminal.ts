import { AnalysisResult, Declaration, DeclarationType } from '$types/analysis';
import { Logger } from '$ui/logger/Logger';
import chalk from 'chalk';

/**
 * Groupe les d√©clarations par type
 */
const groupByType = (
  deadCode: Declaration[],
): Map<DeclarationType, Declaration[]> => {
  const grouped = new Map<DeclarationType, Declaration[]>();

  for (const declaration of deadCode) {
    if (!grouped.has(declaration.type)) {
      grouped.set(declaration.type, []);
    }
    grouped.get(declaration.type)!.push(declaration);
  }

  return grouped;
};

/**
 * Affiche les fonctions non utilis√©es
 */
const displayFunctions = (functions: Declaration[]): void => {
  if (functions.length === 0) return;

  Logger.section('=== Fonctions ===');
  for (const func of functions) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${func.location.filePath}:${func.location.line}:${func.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${func.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les classes non utilis√©es
 */
const displayClasses = (classes: Declaration[]): void => {
  if (classes.length === 0) return;

  Logger.section('=== Classes ===');
  for (const cls of classes) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${cls.location.filePath}:${cls.location.line}:${cls.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${cls.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les constantes non utilis√©es
 */
const displayConstants = (constants: Declaration[]): void => {
  if (constants.length === 0) return;

  Logger.section('=== Constantes ===');
  for (const constant of constants) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${constant.location.filePath}:${constant.location.line}:${constant.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${constant.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les props non utilis√©es
 */
const displayProps = (props: Declaration[]): void => {
  if (props.length === 0) return;

  Logger.section('=== Props ===');
  for (const prop of props) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${prop.location.filePath}:${prop.location.line}:${prop.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${prop.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les console.* non utilis√©s
 */
const displayConsoles = (consoles: Declaration[]): void => {
  if (consoles.length === 0) return;

  Logger.section('=== Console Logs ===');
  for (const cons of consoles) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${cons.location.filePath}:${cons.location.line}:${cons.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Type:')} ${cons.context || cons.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les types/interfaces non utilis√©s
 */
const displayDefinitions = (definitions: Declaration[]): void => {
  if (definitions.length === 0) return;

  Logger.section('=== Types/Interfaces ===');
  for (const def of definitions) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${def.location.filePath}:${def.location.line}:${def.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${def.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les exports non utilis√©s
 */
const displayExports = (exports: Declaration[]): void => {
  if (exports.length === 0) return;

  Logger.section('=== Exports ===');
  for (const exp of exports) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${exp.location.filePath}:${exp.location.line}:${exp.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${exp.name}`);
  }
  Logger.newLine();
};

/**
 * R√©cup√®re les exports utilis√©s uniquement localement (local-only)
 */
const getLocalOnlyExports = (
  declarations: Map<string, Declaration>,
): Declaration[] => {
  const localOnly: Declaration[] = [];

  for (const declaration of declarations.values()) {
    if (
      declaration.type === DeclarationType.EXPORT &&
      declaration.isUsedLocally &&
      !declaration.isImportedExternally
    ) {
      localOnly.push(declaration);
    }
  }

  return localOnly.sort((a, b) =>
    a.location.filePath.localeCompare(b.location.filePath),
  );
};

/**
 * Affiche les exports utilis√©s uniquement localement (sugg√©rer de supprimer 'export')
 */
const displayLocalOnlyExports = (exports: Declaration[]): void => {
  if (exports.length === 0) return;

  Logger.section('=== Exports Local-Only ===');
  Logger.log(
    `${chalk.gray('Suggestion:')} ${chalk.yellow('Supprimer le mot-cl√© "export"')}`,
  );
  for (const exp of exports) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${exp.location.filePath}:${exp.location.line}:${exp.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${exp.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les d√©pendances non utilis√©es
 */
const displayDependencies = (dependencies: Declaration[]): void => {
  if (dependencies.length === 0) return;

  Logger.section('=== Dependencies ===');
  for (const dep of dependencies) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${dep.location.filePath}:${dep.location.line}:${dep.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Name:')} ${dep.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche les d√©clarations marqu√©es @deprecated
 */
const displayDeprecated = (deprecated: Declaration[]): void => {
  if (deprecated.length === 0) return;

  Logger.section('=== D√©clarations @deprecated ===');
  Logger.log(
    `${chalk.gray('Suggestion:')} ${chalk.yellow('Consid√©rer la suppression ou le remplacement')}`,
  );
  for (const decl of deprecated) {
    Logger.log(
      `${chalk.gray('File:')} ${chalk.green(`${decl.location.filePath}:${decl.location.line}:${decl.location.column}`)}`,
    );
    Logger.log(`    ${chalk.gray('Type:')} ${decl.type}`);
    Logger.log(`    ${chalk.gray('Name:')} ${decl.name}`);
  }
  Logger.newLine();
};

/**
 * Affiche le r√©sum√©
 */
const displaySummary = (result: AnalysisResult): void => {
  Logger.section('üìä Code mort d√©tect√©:');

  const groupedByType = groupByType(result.deadCode);

  // Afficher dans l'ordre exact
  const order = [
    { type: DeclarationType.FUNCTION, label: 'Functions' },
    { type: DeclarationType.CLASS, label: 'Classes' },
    { type: DeclarationType.CONSTANT, label: 'Constants' },
    { type: DeclarationType.PROP, label: 'Props' },
    { type: DeclarationType.CONSOLE, label: 'Consoles' },
    { type: DeclarationType.DEFINITION, label: 'Types/Interfaces' },
    { type: DeclarationType.EXPORT, label: 'Exports' },
    { type: DeclarationType.DEPENDENCY, label: 'Dependency' },
  ];

  for (const { type, label } of order) {
    const declarations = groupedByType.get(type);
    if (declarations && declarations.length > 0) {
      Logger.list(`${label}: ${declarations.length}`);
    }
  }

  Logger.newLine();
  Logger.log(`Total: ${result.deadCode.length} √©l√©ments`);

  // Afficher compteur @deprecated si pr√©sent
  if (result.deprecated && result.deprecated.length > 0) {
    Logger.log(
      `‚ö†Ô∏è  D√©clarations ${chalk.yellow('@deprecated')}: ${result.deprecated.length} √©l√©ments`,
    );
  }

  Logger.log(
    `‚è±Ô∏è  Dur√©e totale de l'analyse: ${(result.duration / 1000).toFixed(2)} secondes`,
  );
  Logger.success('Analyse termin√©e');
};

/**
 * Affiche les r√©sultats de l'analyse dans le terminal
 */
export const displayAnalysisResults = (result: AnalysisResult): void => {
  const localOnlyExports = getLocalOnlyExports(result.declarations);

  // Grouper le code mort par type
  const groupedByType = groupByType(result.deadCode);

  // Afficher chaque cat√©gorie dans l'ordre exact
  displayFunctions(groupedByType.get(DeclarationType.FUNCTION) || []);
  displayClasses(groupedByType.get(DeclarationType.CLASS) || []);
  displayConstants(groupedByType.get(DeclarationType.CONSTANT) || []);
  displayProps(groupedByType.get(DeclarationType.PROP) || []);
  displayConsoles(groupedByType.get(DeclarationType.CONSOLE) || []);
  displayDefinitions(groupedByType.get(DeclarationType.DEFINITION) || []);
  displayExports(groupedByType.get(DeclarationType.EXPORT) || []);
  displayLocalOnlyExports(localOnlyExports);
  displayDependencies(groupedByType.get(DeclarationType.DEPENDENCY) || []);

  // Afficher les d√©clarations @deprecated
  if (result.deprecated) {
    displayDeprecated(result.deprecated);
  }

  // R√©sum√©
  displaySummary(result);
};
