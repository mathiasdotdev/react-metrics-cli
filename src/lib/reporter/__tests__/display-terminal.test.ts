import { AnalysisResult, Declaration, DeclarationType } from '$types/analysis';
import chalk from 'chalk';
import { afterEach, beforeEach, describe, expect, it, spyOn } from 'bun:test';
import { displayAnalysisResults } from '../display-terminal';

describe('displayAnalysisResults', () => {
  let consoleLogSpy: ReturnType<typeof spyOn>;
  let originalChalkLevel: typeof chalk.level;

  beforeEach(() => {
    originalChalkLevel = chalk.level;
    chalk.level = 0; // DÃ©sactiver les couleurs
  });

  afterEach(() => {
    if (consoleLogSpy) {
      consoleLogSpy.mockRestore();
    }
    chalk.level = originalChalkLevel;
  });

  /**
   * Capture la sortie console et la retourne comme string pour snapshot
   */
  const captureOutput = (result: AnalysisResult): string => {
    const output: string[] = [];
    consoleLogSpy = spyOn(console, 'log').mockImplementation((...args) => {
      output.push(args.map((arg) => String(arg)).join(' '));
    });

    displayAnalysisResults(result);

    return output.join('\n').replace(/\d+\.\d+\s+secondes?/g, 'X.XX secondes');
  };

  const createDeclaration = (
    name: string,
    type: DeclarationType,
    filePath: string,
    line: number,
    column: number,
    context?: string,
  ): Declaration => ({
    name,
    type,
    location: { filePath, line, column },
    isUsedLocally: false,
    isImportedExternally: false,
    context,
  });

  it('should display empty results', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [],
      deprecated: [],
      duration: 1000,
    };

    const output = captureOutput(result);
    expect(output).toMatchSnapshot();
  });

  it('should display functions only', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [
        createDeclaration(
          'unusedFunction',
          DeclarationType.FUNCTION,
          '/test/file.ts',
          10,
          5,
        ),
        createDeclaration(
          'anotherUnused',
          DeclarationType.FUNCTION,
          '/test/other.ts',
          20,
          3,
        ),
      ],
      duration: 500,
    };

    const output = captureOutput(result);
    expect(output).toMatchSnapshot();
  });

  it('should display classes only', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [
        createDeclaration(
          'UnusedClass',
          DeclarationType.CLASS,
          '/test/class.ts',
          15,
          7,
        ),
        createDeclaration(
          'OldClass',
          DeclarationType.CLASS,
          '/test/legacy.ts',
          30,
          7,
        ),
      ],
      duration: 500,
    };

    const output = captureOutput(result);
    expect(output).toMatchSnapshot();
  });

  it('should display constants and props', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [
        createDeclaration(
          'UNUSED_CONST',
          DeclarationType.CONSTANT,
          '/test/const.ts',
          5,
          7,
        ),
        createDeclaration(
          'unusedProp',
          DeclarationType.PROP,
          '/test/props.tsx',
          8,
          3,
        ),
      ],
      duration: 500,
    };

    const output = captureOutput(result);
    expect(output).toMatchSnapshot();
  });

  it('should display console logs', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [
        createDeclaration(
          'console',
          DeclarationType.CONSOLE,
          '/test/console.ts',
          12,
          3,
          'console.log',
        ),
        createDeclaration(
          'console',
          DeclarationType.CONSOLE,
          '/test/console.ts',
          15,
          3,
          'console.error',
        ),
      ],
      duration: 500,
    };

    const output = captureOutput(result);
    expect(output).toMatchSnapshot();
  });

  it('should display all categories mixed', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [
        createDeclaration('fn', DeclarationType.FUNCTION, '/f.ts', 1, 1),
        createDeclaration('fn2', DeclarationType.FUNCTION, '/f.ts', 2, 1),
        createDeclaration('cls', DeclarationType.CLASS, '/c.ts', 1, 1),
        createDeclaration('const1', DeclarationType.CONSTANT, '/c.ts', 1, 1),
        createDeclaration('const2', DeclarationType.CONSTANT, '/c.ts', 2, 1),
        createDeclaration('const3', DeclarationType.CONSTANT, '/c.ts', 3, 1),
        createDeclaration('prop', DeclarationType.PROP, '/p.ts', 1, 1),
        createDeclaration(
          'console',
          DeclarationType.CONSOLE,
          '/c.ts',
          1,
          1,
          'console.log',
        ),
        createDeclaration('Type1', DeclarationType.DEFINITION, '/t.ts', 1, 1),
        createDeclaration('Export1', DeclarationType.EXPORT, '/e.ts', 1, 1),
        createDeclaration('dep', DeclarationType.DEPENDENCY, '/p.json', 1, 1),
      ],
      duration: 2500,
    };

    const output = captureOutput(result);
    expect(output).toMatchSnapshot();
  });

  it('should display real-world scenario with many elements', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [
        // 10 Functions
        ...Array.from({ length: 10 }, (_, i) =>
          createDeclaration(
            `function${i}`,
            DeclarationType.FUNCTION,
            '/f.ts',
            i + 1,
            1,
          ),
        ),
        // 3 Classes
        ...Array.from({ length: 3 }, (_, i) =>
          createDeclaration(
            `Class${i}`,
            DeclarationType.CLASS,
            '/c.ts',
            i + 1,
            1,
          ),
        ),
        // 12 Constants
        ...Array.from({ length: 12 }, (_, i) =>
          createDeclaration(
            `const${i}`,
            DeclarationType.CONSTANT,
            '/c.ts',
            i + 1,
            1,
          ),
        ),
        // 6 Props
        ...Array.from({ length: 6 }, (_, i) =>
          createDeclaration(
            `prop${i}`,
            DeclarationType.PROP,
            '/p.ts',
            i + 1,
            1,
          ),
        ),
        // 13 Consoles
        ...Array.from({ length: 13 }, (_, i) =>
          createDeclaration(
            'console',
            DeclarationType.CONSOLE,
            '/c.ts',
            i + 1,
            1,
            'console.log',
          ),
        ),
        // 5 Definitions
        ...Array.from({ length: 5 }, (_, i) =>
          createDeclaration(
            `Type${i}`,
            DeclarationType.DEFINITION,
            '/t.ts',
            i + 1,
            1,
          ),
        ),
        // 18 Exports
        ...Array.from({ length: 18 }, (_, i) =>
          createDeclaration(
            `Export${i}`,
            DeclarationType.EXPORT,
            '/e.ts',
            i + 1,
            1,
          ),
        ),
        // 1 Dependency
        createDeclaration(
          'react-dom',
          DeclarationType.DEPENDENCY,
          '/package.json',
          12,
          5,
        ),
      ],
      duration: 2345,
    };

    const output = captureOutput(result);
    expect(output).toMatchSnapshot();
  });
});
