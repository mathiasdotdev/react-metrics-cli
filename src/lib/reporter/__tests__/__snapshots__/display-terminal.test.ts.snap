// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`displayAnalysisResults should display empty results 1`] = `
"üìä Code mort d√©tect√©:


Total: 0 √©l√©ments
‚è±Ô∏è  Dur√©e totale de l'analyse: X.XX secondes
‚úÖ Analyse termin√©e"
`;

exports[`displayAnalysisResults should display functions only 1`] = `
"=== Fonctions ===
File: /test/file.ts:10:5
    Name: unusedFunction
File: /test/other.ts:20:3
    Name: anotherUnused


üìä Code mort d√©tect√©:
- Functions: 2


Total: 2 √©l√©ments
‚è±Ô∏è  Dur√©e totale de l'analyse: X.XX secondes
‚úÖ Analyse termin√©e"
`;

exports[`displayAnalysisResults should display classes only 1`] = `
"=== Classes ===
File: /test/class.ts:15:7
    Name: UnusedClass
File: /test/legacy.ts:30:7
    Name: OldClass


üìä Code mort d√©tect√©:
- Classes: 2


Total: 2 √©l√©ments
‚è±Ô∏è  Dur√©e totale de l'analyse: X.XX secondes
‚úÖ Analyse termin√©e"
`;

exports[`displayAnalysisResults should display constants and props 1`] = `
"=== Constantes ===
File: /test/const.ts:5:7
    Name: UNUSED_CONST


=== Props ===
File: /test/props.tsx:8:3
    Name: unusedProp


üìä Code mort d√©tect√©:
- Constants: 1
- Props: 1


Total: 2 √©l√©ments
‚è±Ô∏è  Dur√©e totale de l'analyse: X.XX secondes
‚úÖ Analyse termin√©e"
`;

exports[`displayAnalysisResults should display console logs 1`] = `
"=== Console Logs ===
File: /test/console.ts:12:3
    Type: console.log
File: /test/console.ts:15:3
    Type: console.error


üìä Code mort d√©tect√©:
- Consoles: 2


Total: 2 √©l√©ments
‚è±Ô∏è  Dur√©e totale de l'analyse: X.XX secondes
‚úÖ Analyse termin√©e"
`;

exports[`displayAnalysisResults should display all categories mixed 1`] = `
"=== Fonctions ===
File: /f.ts:1:1
    Name: fn
File: /f.ts:2:1
    Name: fn2


=== Classes ===
File: /c.ts:1:1
    Name: cls


=== Constantes ===
File: /c.ts:1:1
    Name: const1
File: /c.ts:2:1
    Name: const2
File: /c.ts:3:1
    Name: const3


=== Props ===
File: /p.ts:1:1
    Name: prop


=== Console Logs ===
File: /c.ts:1:1
    Type: console.log


=== Types/Interfaces ===
File: /t.ts:1:1
    Name: Type1


=== Exports ===
File: /e.ts:1:1
    Name: Export1


=== Dependencies ===
File: /p.json:1:1
    Name: dep


üìä Code mort d√©tect√©:
- Functions: 2
- Classes: 1
- Constants: 3
- Props: 1
- Consoles: 1
- Types/Interfaces: 1
- Exports: 1
- Dependency: 1


Total: 11 √©l√©ments
‚è±Ô∏è  Dur√©e totale de l'analyse: X.XX secondes
‚úÖ Analyse termin√©e"
`;

exports[`displayAnalysisResults should display real-world scenario with many elements 1`] = `
"=== Fonctions ===
File: /f.ts:1:1
    Name: function0
File: /f.ts:2:1
    Name: function1
File: /f.ts:3:1
    Name: function2
File: /f.ts:4:1
    Name: function3
File: /f.ts:5:1
    Name: function4
File: /f.ts:6:1
    Name: function5
File: /f.ts:7:1
    Name: function6
File: /f.ts:8:1
    Name: function7
File: /f.ts:9:1
    Name: function8
File: /f.ts:10:1
    Name: function9


=== Classes ===
File: /c.ts:1:1
    Name: Class0
File: /c.ts:2:1
    Name: Class1
File: /c.ts:3:1
    Name: Class2


=== Constantes ===
File: /c.ts:1:1
    Name: const0
File: /c.ts:2:1
    Name: const1
File: /c.ts:3:1
    Name: const2
File: /c.ts:4:1
    Name: const3
File: /c.ts:5:1
    Name: const4
File: /c.ts:6:1
    Name: const5
File: /c.ts:7:1
    Name: const6
File: /c.ts:8:1
    Name: const7
File: /c.ts:9:1
    Name: const8
File: /c.ts:10:1
    Name: const9
File: /c.ts:11:1
    Name: const10
File: /c.ts:12:1
    Name: const11


=== Props ===
File: /p.ts:1:1
    Name: prop0
File: /p.ts:2:1
    Name: prop1
File: /p.ts:3:1
    Name: prop2
File: /p.ts:4:1
    Name: prop3
File: /p.ts:5:1
    Name: prop4
File: /p.ts:6:1
    Name: prop5


=== Console Logs ===
File: /c.ts:1:1
    Type: console.log
File: /c.ts:2:1
    Type: console.log
File: /c.ts:3:1
    Type: console.log
File: /c.ts:4:1
    Type: console.log
File: /c.ts:5:1
    Type: console.log
File: /c.ts:6:1
    Type: console.log
File: /c.ts:7:1
    Type: console.log
File: /c.ts:8:1
    Type: console.log
File: /c.ts:9:1
    Type: console.log
File: /c.ts:10:1
    Type: console.log
File: /c.ts:11:1
    Type: console.log
File: /c.ts:12:1
    Type: console.log
File: /c.ts:13:1
    Type: console.log


=== Types/Interfaces ===
File: /t.ts:1:1
    Name: Type0
File: /t.ts:2:1
    Name: Type1
File: /t.ts:3:1
    Name: Type2
File: /t.ts:4:1
    Name: Type3
File: /t.ts:5:1
    Name: Type4


=== Exports ===
File: /e.ts:1:1
    Name: Export0
File: /e.ts:2:1
    Name: Export1
File: /e.ts:3:1
    Name: Export2
File: /e.ts:4:1
    Name: Export3
File: /e.ts:5:1
    Name: Export4
File: /e.ts:6:1
    Name: Export5
File: /e.ts:7:1
    Name: Export6
File: /e.ts:8:1
    Name: Export7
File: /e.ts:9:1
    Name: Export8
File: /e.ts:10:1
    Name: Export9
File: /e.ts:11:1
    Name: Export10
File: /e.ts:12:1
    Name: Export11
File: /e.ts:13:1
    Name: Export12
File: /e.ts:14:1
    Name: Export13
File: /e.ts:15:1
    Name: Export14
File: /e.ts:16:1
    Name: Export15
File: /e.ts:17:1
    Name: Export16
File: /e.ts:18:1
    Name: Export17


=== Dependencies ===
File: /package.json:12:5
    Name: react-dom


üìä Code mort d√©tect√©:
- Functions: 10
- Classes: 3
- Constants: 12
- Props: 6
- Consoles: 13
- Types/Interfaces: 5
- Exports: 18
- Dependency: 1


Total: 68 √©l√©ments
‚è±Ô∏è  Dur√©e totale de l'analyse: X.XX secondes
‚úÖ Analyse termin√©e"
`;
